lib/src/components/NxTextInput/NxTextInput.scss

```diff
@@ -6,17 +6,18 @@
  */
 @import '../../scss-shared/nx-variables';
 @import '../../scss-shared/_nx-form-validation.scss';
+@import '../../scss-shared/_nx-container-helpers.scss';
 
-.nx-text-input:not(:disabled) {
-  &.nx-text-input--valid {
+.nx-text-input {
+  &.valid:not(.pristine) {
     @include valid();
   }
 
-  &.nx-text-input--invalid {
+  &.invalid {
     @include invalid();
```

Comment:
Red border is missing

Reply:
> Red border is missing

https://github.com/sonatype/sonatype-react-shared-components/pull/57/commits/0da768a36b206b39dff5be5afbac78162d38d69a